cmake/modules/versions.cmake

```diff
@@ -28,24 +32,44 @@ function(_get_git_version _var)
     endif()
 
     # Obtain the closest tag
-    git_describe(dev_version "--always" "--tags" "--abbrev=7" ${ARGN})
-
-    if(dev_version MATCHES "NOTFOUND$")
-        # Fallback version
-        set(dev_version "0.0.0")
+    if (is_driver)
+        git_describe(dev_version "--always" "--tags" "--abbrev=7" --match=*+driver)
     else()
-        # Extract the git version part and make it SemVer friendly (ie. "-1-g02682d7" to "-1+02682d7" )
-        string(REGEX MATCH "(-[1-9][0-9]*-g[0-9a-f][0-9a-f][0-9a-f][0-9a-f][0-9a-f][0-9a-f][0-9a-f])$"
-            git_ver_part "${dev_version}")
-        string(REPLACE "${git_ver_part}" "" dev_version "${dev_version}")
-        string(REPLACE "-g" "+" git_ver_part "${git_ver_part}")
-
-        if(dev_version MATCHES "\\+")
-            string(REPLACE "+" "${git_ver_part}-" dev_version "${dev_version}")
+        git_describe(dev_version "--always" "--tags" "--abbrev=7" --exclude=*+driver)
+    endif()
+    string(REGEX MATCH "^[0-9]+.[0-9]+.[0-9]+$" libs_tag ${dev_version})
+    string(REGEX MATCH "^[0-9]+.[0-9]+.[0-9]+\\+driver$" driver_tag ${dev_version})
+    if(dev_version MATCHES "NOTFOUND$" OR (libs_tag STREQUAL "" AND driver_tag STREQUAL ""))
```

Comment:
Algorithm is the same as on Falco: if we are not on a tag (ie: on a x.y.z or x.y.z+driver), we fetch HEAD, then latest tag (for driver or userspace), and then we mix them together to output a diff-from-latest_tag version.